test/e2e_tests/criticalFlow.spec.ts

```diff
@@ -0,0 +1,307 @@
+    await api.createTeamOwner(owner, teamName);
+    if (!owner.token) {
+      throw new Error(`Owner ${owner.username} has no token and can't be used for team creation`);
+    }
+    const teamId = await api.team.getTeamIdForUser(owner);
+    createdTeams.set(owner, teamId);
+    const invitationId = await api.team.inviteUserToTeam(
+      teamId,
+      member.email,
+      `${owner.firstName} ${owner.lastName}`,
+      owner.token,
+    );
+    const invitationCode = await api.brig.getTeamInvitationCodeForEmail(teamId, invitationId);
+
+    await api.createPersonalUser(member, invitationCode);
+    if (!member.id) {
+      throw new Error(`Member ${member.username} has no ID and can't be invited to the conversation`);
+    }
+    await api.conversation.inviteToConversation(member.id, owner.token, teamId, conversationName);
+  });
+
+  // Test steps
+  await test.step('Members logs in into the application', async () => {
+    await page.goto(webAppPath);
+    await singleSignOnPage.enterEmailOnSSOPage(owner.email);
+    await loginPage.inputPassword(owner.password);
+    await loginPage.clickSignInButton();
+    await dataShareConsentModal.clickDecline();
+  });
+
+  // TODO:
+  // zautomation also features the following steps:
+  //
+  //     When I remember number of reported events
+  //     And User <Owner> pinged in the conversation with Tracking
+  //     Then There are no added reported events
+  //
+  // but these are not a part of the original test case, so I skip them here.
+
+  await test.step('Member opens settings', async () => {
+    await conversationSidebar.clickPreferencesButton();
+  });
+
+  await test.step('Member enables logging in settings', async () => {
+    await accountPage.toggleSendUsageData();
+  });
+
+  await test.step('Member enables applock and sets their password', async () => {
+    await accountPage.toggleAppLock();
+    await appLockModal.setPasscode(appLockPassphrase);
+    await conversationSidebar.clickAllConversationsButton();
+    expect(await conversationListPage.isConversationItemVisible(conversationName));
+  });
+
+  await test.step('Member verifies if applock is working', async () => {
+    await page.reload();
+    expect(await appLockModal.isVisible());
+    expect(await conversationListPage.isConversationItemVisible(conversationName)).toBeFalsy();
+    expect(await appLockModal.getAppLockModalHeader()).toContain('Enter passcode to unlock');
+    expect(await appLockModal.getAppLockModalText()).toContain('Passcode');
+
+    await appLockModal.unlockAppWithPasscode(appLockPassphrase);
+    expect(await appLockModal.isHidden());
+    expect(await conversationListPage.isConversationItemVisible(conversationName));
+  });
+
+  // TODO: Missing test steps for TC-8639 from testiny:
+  // Member changes their email address to a new email address
+  // Member resets their password
+  //
+  // These steps were not implemented in zautomation, so I skipped them here for the time being.
```

Comment:
For the time being fine since we are going for parity, but leave the TODO in so we can address it in a second round